### DISCUSSION TITLE
How to configure lightspeed to work with vim-sandwich?

### DISCUSSION THREAD

**Discussion by Dema** [ggandor/lightspeed.nvim#60] [Q&A] [answered]
Hi!
Lightspeed is using "s" and "S" keys in normal mode which conflicts with vim-sandwich. Can you give an example on how to use your plugin with vim-sandwich by inexperienced user? 

I'm currently stuck, I don't have a lot of experience with vim and I have no idea how configure both plugins to work together. :smile: I love your plugin and vim-sandwich too and I'd like them to be able to work together. Thanks in advance!!

**Comment by ggandor** [comment, marked as answer]
`s` is indeed occupied in Normal and also in Visual mode by vim-sandwich, so if you'd like to keep using sandwich with its default mappings, you should specify custom mappings for Lightspeed in your config:

```Vim
" ? = key you'd like to use
" make sure to use '(n|x)map', and not '(n|x)noremap'
" Normal mode
nmap ? <Plug>Lightspeed_s 
nmap ? <Plug>Lightspeed_S
" Visual mode
xmap ? <Plug>Lightspeed_s 
xmap ? <Plug>Lightspeed_S
```

See also `:h lightspeed-custom-mappings`, and feel free to ask for more help if you're still stuck.

Personally I would use vim-sandwich with [vim-surround keybindings](https://github.com/machakann/vim-sandwich/wiki/Introduce-vim-surround-keymappings), and Lightspeed with the defaults. (In that case the Visual-mode mappings of sandwich should still be tweaked though, because it occupies `S` then.) The thing is, there is no good alternative for `s/S` for Lightspeed in Normal mode. Jumping between an arbitrary A and B position is much more frequent than "sandwiching", so the most comfortable keys should be reserved for the former.
